File: android/app/src/main/java/com/zillit/lcw/data/model/ScheduleEvent.kt
package com.zillit.lcw.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ScheduleEvent(
    @SerialName("_id") val id: String,
    val projectId: String = "",
    val scheduleDayId: String? = null,
    val date: Long = 0L,
    val eventType: String = "note", // "event" or "note"
    var title: String = "",
    var color: String = "#3498DB",
    var description: String = "",
    var startDateTime: Long = 0L,
    var endDateTime: Long = 0L,
    var fullDay: Boolean = false,
    var location: String = "",
    var locationLat: Double? = null,
    var locationLng: Double? = null,
    var reminder: String = "none",
    var repeatStatus: String = "none",
    var repeatEndDate: Long = 0L,
    var timezone: String = "",
    var callType: String = "",
    var textColor: String = "",
    var notes: String = "",
    val deleted: Long = 0L,
    val createdAt: Long = 0L,
    val updatedAt: Long = 0L
)

File: ios/ZillitLCW/Models/ScheduleEvent.swift
import Foundation

struct ScheduleEvent: Identifiable, Codable {
    let id: String
    let projectId: String
    let scheduleDayId: String?
    let date: Int64
    let eventType: String // "event" or "note"
    var title: String
    var color: String
    var description: String?
    var startDateTime: Int64?
    var endDateTime: Int64?
    var fullDay: Bool
    var location: String?
    var locationLat: Double?
    var locationLng: Double?
    var reminder: String
    var repeatStatus: String
    var repeatEndDate: Int64?
    var timezone: String?
    var callType: String?
    var textColor: String?
    var notes: String?
    var distributeTo: String?
    var distributeUserIds: [String]?
    var distributeDepartmentIds: [String]?
    var userPresetId: String?
    var organizerExcluded: Bool?
    let deleted: Int64
    let createdAt: Int64
    let updatedAt: Int64

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case projectId, scheduleDayId, date, eventType, title, color
        case description, startDateTime, endDateTime, fullDay
        case location, locationLat, locationLng
        case reminder, repeatStatus, repeatEndDate, timezone, callType, textColor
        case notes
        case distributeTo, distributeUserIds, distributeDepartmentIds, userPresetId, organizerExcluded
        case deleted, createdAt, updatedAt
    }
}
